src/sentry/event_manager.py

```diff
@@ -1935,6 +1944,123 @@ def _detect_performance_problems(jobs, projects):
+
+        # General system-wide option
+        rate = options.get("performance.issues.all.problem-creation") or 0
+
+        # More granular, per-project option
+        per_project_rate = project.get_option("sentry:performance_issue_creation_rate", 0)
+        if rate > random.random() and per_project_rate > random.random():
+
+            kwargs = _create_kwargs(job)
+            kwargs["culprit"] = job["culprit"]
+            kwargs["data"] = materialize_metadata(
+                event.data,
+                get_event_type(event.data),
+                dict(job["event_metadata"]),
+            )
+            kwargs["data"]["last_received"] = job["received_timestamp"]
+
+            performance_problems = job["performance_problems"]
+            for problem in performance_problems:
+                problem.fingerprint = md5(problem.fingerprint.encode("utf-8")).hexdigest()
+
+            performance_problems_by_fingerprint = {p.fingerprint: p for p in performance_problems}
+            all_group_hashes = [problem.fingerprint for problem in performance_problems]
+            group_hashes = all_group_hashes[:MAX_GROUPS]
+
+            existing_grouphashes = GroupHash.objects.filter(
+                project=project, hash__in=group_hashes
+            ).select_related("group")
+
+            new_grouphashes = set(group_hashes) - {hash.hash for hash in existing_grouphashes}
+            new_grouphashes_count = len(new_grouphashes)
+
+            if new_grouphashes:
+                granted_quota = issue_rate_limiter.check_and_use_quotas(
+                    [
+                        RequestedQuota(
+                            f"performance-issues:{project.id}",
+                            new_grouphashes_count,
+                            [PERFORMANCE_ISSUE_QUOTA],
+                        )
+                    ]
+                )[0]
+
+                # Log how many groups didn't get created because of rate limiting
+                _dropped_group_hash_count = new_grouphashes_count - granted_quota.granted
+                metrics.incr("performance.performance_issue.dropped", _dropped_group_hash_count)
+
+                for new_grouphash in list(new_grouphashes)[: granted_quota.granted]:
+
+                    # GROUP DOES NOT EXIST
+                    with sentry_sdk.start_span(
+                        op="event_manager.create_group_transaction"
+                    ) as span, metrics.timer(
+                        "event_manager.create_group_transaction"
+                    ) as metric_tags, transaction.atomic():
+                        span.set_tag("create_group_transaction.outcome", "no_group")
+                        metric_tags["create_group_transaction.outcome"] = "no_group"
+
+                        problem = performance_problems_by_fingerprint[new_grouphash]
+                        kwargs["type"] = problem.type.value
+                        kwargs["data"]["metadata"]["title"] = f"N+1 Query:{problem.desc}"
```

Comment:
Oh! Maybe the problem detectors should return a string for the problem type?

Reply:
yeah that'd work here better 👍 gonna hard-code it for now since we're only returning one type of perf problems

Reply:
I think we should just have a mapping of `GroupType` -> `str` somewhere, since the `PerformanceProblem` already has that reference. Performance detection doesn't really need to know how we display a given performance problem